solutions/scaling_microfrontend_aws_eks/index.asciidoc

```diff
@@ -0,0 +1,161 @@
+1. Dockerize your micro apps +
+2. Push your docker images to a container registry +
+3. Create your Cluster and Worker Nodes +
+4. Configure kubectl to use Cluster +
+5. Deploy your apps on AWS EKS Cluster +
+
+=== Dockerize your micro apps
+
+You start with dockerizing your micro apps by creating a Dockerfile for each of them. Your Dockerfiles will look like the following if the apps are developed in Angular:
+
+```
+FROM node AS ui-build
+WORKDIR /usr/src/app
+COPY micro-app-1/ ./micro-app-1/
+RUN cd micro-app-1 && npm install @angular/cli && npm install && npm run build
+
+
+FROM nginx:alpine
+
+#!/bin/sh
+
+COPY ./.nginx/nginx.conf /etc/nginx/nginx.conf
+
+## Remove default nginx index page
+RUN rm -rf /usr/share/nginx/html/*
+
+# Copy from the stahg 1
+COPY --from=ui-build /usr/src/app/micro-app-1/dist/ /usr/share/nginx/html
+
+EXPOSE 4200 80
+
+ENTRYPOINT ["nginx", "-g", "daemon off;"]
+
+```
+
+Now you build your docker images using the `docker build` command.
+
+=== Push docker images to a container registry
+
+Once you have your docker images, you need to push them to a container registry. A container registry serves as a repository for your docker images. AWS Elastic Container Service is one such container registry.
+
+You can create a repository in AWS ECS either using the console or the AWS CLI. Follow https://docs.aws.amazon.com/AmazonECR/latest/userguide/repository-create.html[this guide] to create a reposistory in ECS.
+
+Once you have your repository, tag your docker images with the repository URI using the `docker tag` command, and then push the image to the reposiroty using `docker push`.
+
+=== Create your Cluster and Worker Nodes
+
+Now that your docker images are in your repository, you are ready to https://docs.aws.amazon.com/eks/latest/userguide/create-cluster.html[create your cluster] followed by the worker nodes and the https://docs.aws.amazon.com/eks/latest/userguide/create-managed-node-group.html[node group]. Keep in mind that you also need to create appropriate roles for the cluster and the nodes. You can also use `eksctl` to create a cluster and a node group. `eksctl` abstracts away a lot of the steps need to manually create clusters, node groups, etc.
+
+=== Configure kubectl to use Cluster
+
+Once your cluster is setup, you can use the `update-kubeconfig` command of AWS CLI to connect to your cluster. As a pre-requistite to this step, you need to https://docs.aws.amazon.com/eks/latest/userguide/install-aws-iam-authenticator.html[authenticate the cluster]. +
+The `update-kubeconfig` command should look like:
+```
+aws eks --region <region-code> update-kubeconfig --name <cluster-name>
+```
+
+=== Deploy your apps on AWS EKS Cluster
+Finally, you deploy your micro apps on your EKS cluster. The configuration file for your individual micro app should look something like this:
+
+```
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  labels:
+    app: micro-app-1
+  name: micro-app-1-deployment
+spec:
+  replicas: 5
```

Comment:
I do not understan why do we need to scale the instance of a frontend application (even a microfrontend) if it executes on users browser. Scalation itself is related to the microfrontend approach!